src/model_io/urdf-kdl/src/urdf_import.cpp

```diff
@@ -377,6 +377,14 @@ bool framesFromKDLTree(const KDL::Tree& tree,
             std::string parentLinkName = GetTreeElementSegment(GetTreeElementParent(seg->second)->second).getName();
             framesNames.push_back(frameName);
             parentLinkNames.push_back(parentLinkName);
+
+            //also check parent
+            KDL::Segment parent = GetTreeElementSegment(GetTreeElementParent(seg->second)->second);
+            if (parent.getJoint().getType() == KDL::Joint::None &&
+                parent.getInertia().getMass() == 0.0)
+            {
+                framesNames.push_back(parentLinkName);
+            }
```

Comment:
Basically if you have two consecutive massless links, connected by two fixed joints, consider both of them "frames"? This make sense, I noticed that you have some of those in the Walkman model.